beacon_node/network/src/beacon_processor/worker/gossip_methods.rs

```diff
@@ -291,19 +291,34 @@ impl<T: BeaconChainTypes> Worker<T> {
         reprocess_tx: mpsc::Sender<ReprocessQueueMessage<T>>,
         seen_duration: Duration,
     ) {
+        let block_delay =
+            get_block_delay_ms(seen_duration, block.message(), &self.chain.slot_clock);
         // Log metrics to track delay from other nodes on the network.
         metrics::observe_duration(
             &metrics::BEACON_BLOCK_GOSSIP_SLOT_START_DELAY_TIME,
-            get_block_delay_ms(seen_duration, block.message(), &self.chain.slot_clock),
+            block_delay,
         );
 
         let verified_block = match self.chain.verify_block_for_gossip(block) {
             Ok(verified_block) => {
+                if block_delay >= self.chain.slot_clock.unagg_attestation_production_delay() {
+                    metrics::inc_counter(&metrics::BEACON_BLOCK_GOSSIP_ARRIVED_LATE_TOTAL);
+                    debug!(
+                        self.log,
+                        "Gossip block arrived late";
```

Comment:
Similarly, there's already an equivalent to this log message here:

https://github.com/sigp/lighthouse/blob/12fe72bd372b6ffc788572aa0b3c559961a97a55/beacon_node/beacon_chain/src/beacon_chain.rs#L2958-L2969

Reply:
Oh yeah, this one is slightly different though.

My new log tells me if the block arrived late *on gossip*, whilst the other one tells me if it was late *to be set as head*. With the new log I can now tell if it arrived late or if we were too slow to import it/set it as head. #2528 details a better solution to this, but this new log gets it done for the time being.

Now you mention it I might add the proposer index to the existing log (the one you referenced here). Thanks!

Reply:
Oh yeah, nice. That's subtle but useful